src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -149,6 +150,12 @@ public override void VisitCSharpImplicitExpression(CSharpImplicitExpressionSynta
             Visit(node.Body);
         }
 
+        public override void VisitCSharpImplicitExpressionBody(CSharpImplicitExpressionBodySyntax node)
+        {
+            AddSemanticRange(node.CSharpCode, RazorSemanticTokensLegend.RazorCodeBackground);
+            Visit(node.CSharpCode);
```

Comment:
Hmm, for clarity why not override the VisitCSharpCodeBlock method and add the background coloring there?